test/unittests/execute_floating_point_test.cpp

```diff
@@ -66,4 +69,203 @@ TEST(execute_floating_point, f64_add)
+    static constexpr auto dst_valtype = ValType::i64;
+};
+
+template <typename SrcT, typename DstT>
+struct ConversionPair : ConversionPairWasmTraits<SrcT, DstT>
+{
+    using src_type = SrcT;
+    using dst_type = DstT;
+};
+
+struct ConversionName
+{
+    template <typename T>
+    static std::string GetName(int /*unused*/)
+    {
+        return T::opcode_name;
+    }
+};
+
+template <typename T>
+class execute_floating_point_trunc : public testing::Test
+{
+};
+
+using TruncPairs = testing::Types<ConversionPair<float, int32_t>, ConversionPair<float, uint32_t>,
+    ConversionPair<double, int32_t>, ConversionPair<double, uint32_t>,
+    ConversionPair<float, int64_t>, ConversionPair<float, uint64_t>,
+    ConversionPair<double, int64_t>, ConversionPair<double, uint64_t>>;
+TYPED_TEST_SUITE(execute_floating_point_trunc, TruncPairs, ConversionName);
+
+TYPED_TEST(execute_floating_point_trunc, trunc)
+{
+    using FloatT = typename TypeParam::src_type;
+    using IntT = typename TypeParam::dst_type;
+    using FloatLimits = std::numeric_limits<FloatT>;
+    using IntLimits = std::numeric_limits<IntT>;
+
+    /* wat2wasm
+    (func (param f32) (result i32)
+      local.get 0
+      i32.trunc_f32_s
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001060160017d017f030201000a070105002000a80b");
+
+    // Find and replace changeable values: types and the conversion instruction.
+    auto& param_type = wasm[wasm.find(static_cast<uint8_t>(ValType::f32))];
+    auto& result_type = wasm[wasm.find(static_cast<uint8_t>(ValType::i32))];
+    auto& opcode = wasm[wasm.find(static_cast<uint8_t>(Instr::i32_trunc_f32_s))];
+    param_type = static_cast<uint8_t>(TypeParam::src_valtype);
+    result_type = static_cast<uint8_t>(TypeParam::dst_valtype);
+    opcode = static_cast<uint8_t>(TypeParam::opcode);
+
+    auto instance = instantiate(parse(wasm));
+
+    // Zero.
+    EXPECT_THAT(execute(*instance, 0, {FloatT{0}}), Result(IntT{0}));
+    EXPECT_THAT(execute(*instance, 0, {-FloatT{0}}), Result(IntT{0}));
+
+    // Something around 0.0.
+    EXPECT_THAT(execute(*instance, 0, {FloatLimits::denorm_min()}), Result(IntT{0}));
+    EXPECT_THAT(execute(*instance, 0, {-FloatLimits::denorm_min()}), Result(IntT{0}));
+
+    // Something smaller than 2.0.
+    EXPECT_THAT(execute(*instance, 0, {std::nextafter(FloatT{2}, FloatT{0})}), Result(IntT{1}));
+
+    // Something bigger than -1.0.
+    EXPECT_THAT(execute(*instance, 0, {std::nextafter(FloatT{-1}, FloatT{0})}), Result(IntT{0}));
+
+    {
+        // BOUNDARIES OF DEFINITION
+        //
+        // Here we want to identify and test the boundary values of the defined behavior of the
+        // trunc instructions. For undefined results the execution must trap.
+        // Note that floating point type can represent any power of 2.
+
+        using expected_boundaries = trunc_boundaries<FloatT, IntT>;
+
+        // For iN with max value 2^N-1 the float(2^N) exists and trunc() from that value to iN
```

Comment:
To disambiguate from "max value" in the beginning of the sentence
```suggestion
        // For iN with max value 2^N-1 the float(2^N) exists and trunc(float(2^N)) to iN
```